### ISSUE TITLE
Release weak modifiers on/before another key gets pressed?

### ISSUE THREAD

**Issue by girst** [tmk/tmk_keyboard#534]
I've got the following situation: I'm using an `ACTION_FUNCTION` on a key I'm calling slash-question. 

In my normal German layout, the slash is on Shift-7, and the question mark on Shift-hyphen. My function looks like this:

```c
void action_function(keyrecord_t *record, uint8_t id, uint8_t opt) {
        uint8_t shifted = get_mods() & (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT));
        static uint8_t slqu_pressed; //for SLASH_QUESTION

        switch (id) {
        case SLASH_QUESTION:
                if (record->event.pressed) {
                        slqu_pressed = shifted ? KC_MINS : KC_7;
                        add_weak_mods(MOD_BIT(KC_LSHIFT)), send_keyboard_report();
                        add_key(slqu_pressed),             send_keyboard_report();
                } else {
                        del_key(slqu_pressed),             send_keyboard_report();
                        del_weak_mods(MOD_BIT(KC_LSHIFT)), send_keyboard_report();
                }
                break;
        }
}
```

what often happens is I want to type `cd /tmp`, but I'll press the `t` before I release `/`, which would work if this were implemented in the keyboard layout, but using above function it'll type `cd /Tmp`.     
This issue also happens with all my `ACTION_MODS_KEY(MOD_RALT, KC_something)` mappings (which doesn't bother me too much).

**TL;DR:** Is there a way to make the firmware release modifiers set with `add_weak_mods` when any other key than this one is pressed?

[My firmware is here](https://github.com/girst/tmk_keyboard/blob/master/keyboard/tv44/keymap_tv44.c), if that helps. 

quick addition: I thought, maybe `record->tap.interrupted` could be used to detect when another key gets pressed and release the weak mods, but that didn't fire.

**Comment by tmk** [comment]
You want to `del_weak_mods`(but won't need `send_keyboard_report()` ) just after sending SHIFT and key. With this when you press 't' it sends 't' without SHIFT.


This will cure your problem in most cases, there may be OS/host which refuse this but who cares.

**Comment by girst** [comment]
Thank you very much, that works a treat!

for future reference, here's the modified code: 

```
        case SLASH_QUESTION:
                if (record->event.pressed) {
                        slqu_pressed = shifted?KC_MINS:KC_7;

                        add_weak_mods(MOD_BIT(KC_LSHIFT));
                        add_key(slqu_pressed);
                        send_keyboard_report();
                        del_weak_mods(MOD_BIT(KC_LSHIFT));
                } else {
                        del_key(slqu_pressed);
                        send_keyboard_report();
                }
                break;
```
